oauth2_provider/oauth2_validators.py

```diff
@@ -277,6 +277,11 @@ def _get_token_from_authentication_server(
             log.exception("Introspection: Failed POST to %r in token lookup", introspection_url)
             return None
 
+        if response.status_code == 403:
```

Comment:
You should check against the expected status code rather.

Reply:
ok, but then we can't say that we failed to authenticate or permissions are wrong (401, 403), "if not status_code == 200", is that ok?

Reply:
If you want to handle specific status codes that is fine. You should have an else clause in there though if you do.

Reply:
BTW a couple of tests would be nice for this.